src/components/capture/TestButton.tsx

```diff
@@ -0,0 +1,142 @@
+import { Button } from '@mui/material';
+import { discover } from 'api/discovers';
+import { createEntityDraft } from 'api/drafts';
+import { encryptConfig } from 'api/sops';
+import { EditorStoreState } from 'components/editor/Store';
+import { buttonSx } from 'components/shared/Entity/Header';
+import { DraftSpecQuery } from 'hooks/useDraftSpecs';
+import { useRouteStore } from 'hooks/useRouteStore';
+import { useZustandStore } from 'hooks/useZustand';
+import { isEmpty } from 'lodash';
+import { FormattedMessage } from 'react-intl';
+import {
+    entityCreateStoreSelectors,
+    formInProgress,
+    FormStatus,
+} from 'stores/Create';
+
+interface Props {
+    disabled: boolean;
+    formId: string;
+    onFailure: Function;
+    subscription: Function;
+}
+
+function CaptureTestButton({
+    disabled,
+    formId,
+    onFailure,
+    subscription,
+}: Props) {
+    const draftId = useZustandStore<
+        EditorStoreState<DraftSpecQuery>,
+        EditorStoreState<DraftSpecQuery>['id']
+    >((state) => state.id);
+
+    const entityCreateStore = useRouteStore();
+    const formStateStatus = entityCreateStore(
+        entityCreateStoreSelectors.formState.status
+    );
+    const setFormState = entityCreateStore(
+        entityCreateStoreSelectors.formState.set
+    );
+
+    const entityName = entityCreateStore(
+        entityCreateStoreSelectors.details.entityName
+    );
+    const imageTag = entityCreateStore(
+        entityCreateStoreSelectors.details.connectorTag
+    );
+    const endpointConfigData = entityCreateStore(
+        entityCreateStoreSelectors.endpointConfig.data
+    );
+    const endpointSchema = entityCreateStore(
+        entityCreateStoreSelectors.endpointSchema
+    );
+    const [detailErrors, specErrors] = entityCreateStore(
+        entityCreateStoreSelectors.errors
+    );
+
+    const test = async (event: React.MouseEvent<HTMLElement>) => {
+        event.preventDefault();
+        setFormState({
+            status: FormStatus.TESTING,
+        });
+
+        let detailHasErrors = false;
+        let specHasErrors = false;
+
+        // TODO (linting) - this was to make TS/Linting happy
+        detailHasErrors = detailErrors ? detailErrors.length > 0 : false;
+        specHasErrors = specErrors ? specErrors.length > 0 : false;
+
+        if (isEmpty(endpointConfigData) || detailHasErrors || specHasErrors) {
+            setFormState({
+                status: FormStatus.IDLE,
+                displayValidation: true,
+            });
+        } else {
+            const draftsResponse = await createEntityDraft(entityName);
+            if (draftsResponse.error) {
+                return onFailure({
+                    error: {
+                        title: 'captureCreation.test.failedErrorTitle',
+                        error: draftsResponse.error,
+                    },
+                });
+            }
```

Comment:
No longer having callback hell. We just await and then check if there was an error.